LibraryApp/src/components/books-list.js

```diff
@@ -0,0 +1,240 @@
+const {p10, py20, px10, px15, px25, p15, mtAuto, mb15} = Spacing;
+const {blue, grey, darkGrey, purple, lightGreen, maroon, white} = Colors;
+
+class BooksList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDataFetching: false,
+      isRefreshing: false,
+      searchText: '',
+    };
+    this.onRefresh = this.onRefresh.bind(this);
+    this.getApiData = this.getApiData.bind(this);
+    this.selectBook = this.selectBook.bind(this);
+    this.search = this.search.bind(this);
+    this.addNewBook = this.addNewBook.bind(this);
+  }
+
+  componentDidMount() {
+    const {books} = this.props;
+    if (books.length == 0) {
+      this.setState({
+        isDataFetching: true,
+      });
+    }
+    setTimeout(() => {
+      this.setState({isDataFetching: false});
+    }, loaderTimeout);
+  }
+
+  onRefresh() {
+    this.setState({isRefreshing: true}, () => {
+      this.getApiData();
+    });
+  }
+
+  getApiData() {
+    setTimeout(() => {
+      this.setState({isRefreshing: false});
+    }, loaderTimeout);
+  }
+
+  selectBook(book) {
+    const {setSelectedBook} = this.props;
+    setSelectedBook(book);
+    this.navigateToForm();
+  }
+
+  renderItem = ({item}) => {
+    const {name, author, publisher, price} = item;
+    const {listItemContainer, listItemName, bookPrice} = styles;
+    return (
+      <Pressable
+        style={listItemContainer}
+        onPress={() => this.selectBook(item)}>
+        <View style={[flex1, px15]}>
+          <Text style={listItemName}>{name}</Text>
+          <Text style={[gochiFont, fs20]}>{author}</Text>
+          <Text style={[gochiFont, fs18]}>{publisher}</Text>
+        </View>
+        <View>
+          <Text style={bookPrice}>{price}</Text>
+        </View>
+      </Pressable>
+    );
+  };
+
+  getItemKey = (item) => `book-${item.id}`;
+
+  search(value) {
+    this.setState({
+      searchText: value.trim(),
+    });
+  }
+
+  addNewBook() {
+    const {setSelectedBook} = this.props;
+    setSelectedBook();
+    this.navigateToForm();
+  }
+
+  navigateToForm() {
+    const {setActiveTab} = this.props;
+    setActiveTab(libraryForm);
+  }
+
+  render() {
+    const {books} = this.props;
+    const {isDataFetching, isRefreshing, searchText} = this.state;
+    const {
+      titleContainer,
+      title,
+      searchContainer,
+      searchIcon,
+      listContainer,
+      button,
+    } = styles;
+    const booksArr = Object.values(books);
+    const filteredList = FilterListBySearch(searchText, booksArr);
+    const containsBooks = filteredList.length > 0;
+    return (
+      <>
+        <View style={titleContainer}>
+          <Text style={title}>List of the Books</Text>
+        </View>
+        <View style={p15}>
+          <View style={searchContainer}>
+            <View style={[verticalCenter]}>
+              <Text style={searchIcon}>{search}</Text>
+            </View>
+            <TextInput
```

Comment:
add a ReturnKeyType for input fields

Reply:
Added returnKeyType="search" but searching is working on entering search text only.